module/network/RobotCommunication/src/RobotCommunication.cpp

```diff
@@ -139,11 +143,49 @@ namespace module::network {
                                 }
                             });
                 }
+
+                emit(std::unique_ptr<TeamMates>());
             });
 
+
+        // include teammates and add a using teammates
+        on<Trigger<RoboCup>, With<TeamMates>>().then([this](const RoboCup& robocup, const TeamMates& old_teammates) {
+            // Get the id of this robot
+            u_int32_t id = robocup.current_pose.player_id;
+
+            // Make new TeamMates message using data from the old message
+            TeamMates teammates = old_teammates;
+
+            // Loop through the teammates vector to check if any teammate ID matches given ID.
+            bool found = false;
+
+            for (auto& mate : teammates.teammates) {  // Loop through each teammate.
+                if (mate.id == id) {  // If the ID of a teammate is found to be the same, update the position.
+                    mate.rRFf = robocup.current_pose.position.cast<double>();  // Update position
+                    found     = true;
+                    break;  // Break out of the loop.
+                }
+            }
+
+            // Else this robot is not in the list of team mates already, add it in
+            if (!found) {
+                // Make the mate
+                TeamMate mate;
+                // Set the variables
+                mate.id   = id;
+                mate.rRFf = robocup.current_pose.position.cast<double>();
+                // Add it to the teammates list
+                teammates.teammates.emplace_back(mate);
+            }
+
+            // After updating or adding a teammate,
+            // move the updated teammate list into a unique pointer and emit it.
+            emit(std::make_unique<TeamMates>(std::move(teammates)));
+        }
```

Comment:
```suggestion
        }

```
add a line in to separate the reactions